simple_rl/agents/func_approx/dqn/DQNAgentClass.py

```diff
@@ -506,6 +503,26 @@ def __init__(self, action_size, buffer_size, batch_size, seed, device):
 
         self.positive_transitions = []
 
+    def __getstate__(self):
+        excluded_keys = ("experience", "memory")
+        state_dictionary = {x: self.__dict__[x] for x in self.__dict__ if x not in excluded_keys}
+
+        # need to convert to tuples because can't pickle NamedTuple
+        memory = [tuple(exp) for exp in self.memory]
+        state_dictionary["memory"] = memory
+        return state_dictionary
+
+    def __setstate__(self, state_dictionary):
```

Comment:
what about the policy and target networks? How are they being get and set?

Reply:
This is the setter for `ReplayBuffer`, not the setter for `DQNAgentClass`. The policy networks are being set on line 481.

Also, let me know if there is a better way to reference another line. :)